pythonopensubtitles/opensubtitles.py

```diff
@@ -104,9 +109,99 @@ def detect_language(self):
         # array DetectLanguage( $token, array($text, $text, ...) )
         raise NotImplementedError
 
-    def download_subtitles(self):
-        # array DownloadSubtitles( $token, array($IDSubtitleFile, $IDSubtitleFile,...) )
-        raise NotImplementedError
+    def download_subtitles(self, subfile_ids, ext='srt',
+                           filenames=None, directory=None):
+        """
+        Returns a dictionary with max. 20 IDs of and paths to all
+        successfully downloaded subtitle files, otherwise None.
+
+        subfile_ids can be None/empty if a filenames dict is present;
```

Comment:
I wouldn't make this optional. Actually, it's not that optional, it needs to receive a None.

What do you think about a little bit of renaming:

```
def download_subtitles(self, ids, override_filenames=None, output_directory='.', extension='srt')
```

But even like that I think that the `ids` should always be mandatory. You will see an example later on.

Also, I think that the extension bit is not needed, we could achieve the same with `override_filenames` or later on with a `mv`.

Reply:
I like the new attribute names!